homework-submission/week4/my-app/src/components/mainInfo.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import getMainInfo from '../services/GetMainInfo';
+
+class MainInfo extends React.Component {
+  state = {
+    text: '',
+    isLiked: true,
+    editMode: false,
+    author: {
+      firstName: '',
+      lastName: '',
+      avatarURL: 'http://www.europe-together.eu/wp-content/themes/sd/images/user-placeholder.svg',
+    },
+  };
+  onDelete = id => {
+    getMainInfo.deleteItem(id).then(data => {
+      this.props.onItemRemoved(id);
+    });
+  };
+
+  onLike = (id, info) => {
+    info.isLiked = !info.isLiked;
+    getMainInfo.checkLiked(id, info).then(newInfo => {
+      this.props.onItemLiked(id, newInfo);
+    });
+  };
+
+  onEdit = (id, info) => {
+    info.editMode = true;
+    this.setState({ editMode: info.editMode });
+  };
+
+  onCancel = (id, info) => {
+    info.editMode = false;
+    this.setState({ editMode: info.editMode });
+  };
+
+  onConfirm = (id, info) => {
+    info.text = this.state.text;
+    info.isLiked = this.state.isLiked;
+    info.author = this.state.author;
```

Comment:
info is the data which is coming from the state in App, why do you change it here, it is risky